verification/ci-verify.py

```
@@ -0,0 +1,1555 @@
+                print(f"       [compare] variant non-zero exit: {res_var.returncode}")
+                print(f"       stderr: {res_var.stderr[:300]}")
+                return False
+
+            base_file = _find_first_output_file(base_dir)
+            var_file  = _find_first_output_file(var_dir)
+
+            if base_file is None or var_file is None:
+                print(f"       [compare] missing output file (base={base_file}, var={var_file})")
+                return False
+
+            base_content = open(base_file, "rb").read()
```

Ruff (0.15.17)

[warning] Use a context manager for opening files

(SIM115)